Source/common/SNTRule.m

```diff
@@ -48,6 +49,57 @@ - (instancetype)initWithIdentifier:(NSString *)identifier
   return self;
 }
 
+// Converts rule information downloaded from the server into a SNTRule.  Because any information
+// not recorded by SNTRule is thrown away here, this method is also responsible for dealing with
+// the extra bundle rule information (bundle_hash & rule_count).
+- (instancetype)initWithDictionary:(NSDictionary *)dict {
```

Comment:
We should write tests for this since we're kinda relying on it to filter out malformed config data.